src/main/scala/analysis/SymbolicAccessAnalysis.scala

```diff
@@ -0,0 +1,152 @@
+package analysis
+
+import analysis.BitVectorEval.{bv2SignedInt, isNegative}
+import analysis.solvers.ForwardIDESolver
+import ir.IRWalk.procedure
+import ir.{Assign, BVADD, BinaryExpr, BitVecLiteral, BitVecType, CFGPosition, DirectCall, Expr, Extract, GoTo, IndirectCall, Literal, Memory, MemoryLoad, Procedure, Program, Register, Repeat, SignExtend, UnaryExpr, Variable, ZeroExtend}
+
+import java.math.BigInteger
+
+case class SymbolicAccess(accessor: Variable, symbolicBase: MemoryLocation, offset: BigInt) {
+  override def toString: String = s"SymbolicAccess($accessor, $symbolicBase, $offset)"
+}
```

Comment:
These probably need to include the size of the access as well to give fully correct results

Reply:
this is a reaching definitions + constant propagation combined

all this track looks something like this
cfgposition -> [(r0 == R31 + c) -> Top],

it conveys that at position cfgposition, R0 may be an alias for R31 + C where C is a known constant. 
Note that not all accesses have size associated with them (e.g. in the example above r0 or R31 + c may never be used as index into memory (in memory read or write with a size). 

I think what may confusing here is the they are named SymbolicAccess but it is really a Symbolic Address not an access

Reply:
That makes sense, thanks, maybe just rename it to 'SymbolicAddress' then?